ta/pkcs11/src/pkcs11_helpers.c

```diff
@@ -479,6 +479,42 @@ bool pkcs2tee_load_attr(TEE_Attribute *tee_ref, uint32_t tee_id,
 	return true;
 }
 
+/*
+ * Initialize a TEE attribute with hash of a target PKCS11 TA attribute
+ * in an object.
+ */
+bool pkcs2tee_load_hashed_attr(TEE_Attribute *tee_ref, uint32_t tee_id,
+			       struct pkcs11_object *obj,
+			       enum pkcs11_attr_id pkcs11_id, uint32_t tee_algo,
+			       void *hash_ptr, uint32_t *hash_size)
+{
+	TEE_OperationHandle handle = TEE_HANDLE_NULL;
+	void *a_ptr = NULL;
+	uint32_t a_size = 0;
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	if (get_attribute_ptr(obj->attributes, pkcs11_id, &a_ptr, &a_size))
+		return false;
+
+	res = TEE_AllocateOperation(&handle, tee_algo, TEE_MODE_DIGEST, 0);
+	if (res) {
+		EMSG("TEE_AllocateOperation() failed %#"PRIx32, tee_algo);
+		return false;
```

Comment:
i think here it would be nice to get a out-of-mem status so I think returning an `enum pkcs11_rc` would be better.

Reply:
Changed the return type back to enum pkcs11_rc